### ISSUE TITLE
Questions about running the registration code

### ISSUE THREAD

**Issue by BhLee98** [neel-dey/anatomix#1]
Hi, I have used the AbdomenMRCT dataset and tried to run your alignment code but did not get the desired results, is there a problem with my settings? My settings are as follows:
```
    result_path = './results'
        convex_adam(
            ckpt_path='../model-weights/best_val_net_G.pth',
            expname='demo',
            lambda_weight=0.75,
            grid_sp=2,
            disp_hw=1,
            selected_niter=80,
            selected_smooth=0,
            grid_sp_adam=2,
            ic=True,
            result_path=result_path,
            fixed_image=fixed_image,
            moving_image=moving_image,
            use_mask=False,
            # fixed_mask=fixed_mask,  # 提供 fixed mask
            # moving_mask=moving_mask,  # 提供 moving mask
            fixed_minclip=-450,
            fixed_maxclip=450,
        )
```

Thanks.

**Comment by neel-dey** [comment]
Hi @BhLee98,

Thanks for your interest! 

The AbdomenMRCT experiments used the ROI masks provided by the dataset/challenge (in `masksTr`). Set `use_mask=True` and provide the mask paths. Without the masks, all the hyperparameters for the underlying convexadam solver would need to be reoptimized with another gridsearch. 

Also, I just pushed an addition to the CLI to optionally warp a segmentation label map and compute Dice for convenience, please pull the latest commit.

To get our results, your calls should look like:
```
python run_convex_adam_with_network_feats.py \
--ckpt_path ../../model-weights/anatomix.pth \
--exp_name withmask \
--fixed_minclip -450 \
--fixed_maxclip 450 \
--result_path ./whereveryouwant/ \
--use_mask \
--warp_seg \
--fixed ./learn2reg-datasets/AbdomenMRCT/imagesTr/AbdomenMRCT_0001_0001.nii.gz \
--moving ./learn2reg-datasets/AbdomenMRCT/imagesTr/AbdomenMRCT_0001_0000.nii.gz \
--path_mask_fixed ./learn2reg-datasets/AbdomenMRCT/masksTr/AbdomenMRCT_0001_0001.nii.gz \
--path_mask_moving ./learn2reg-datasets/AbdomenMRCT/masksTr/AbdomenMRCT_0001_0000.nii.gz \
--path_seg_fixed ./learn2reg-datasets/AbdomenMRCT/labelsTr/AbdomenMRCT_0001_0001.nii.gz \
--path_seg_moving ./learn2reg-datasets/AbdomenMRCT/labelsTr/AbdomenMRCT_0001_0000.nii.gz
```

That should fix it for you :)

**Comment by neel-dey** [comment]
Hi @BhLee98,

In case it's helpful, there's now also a [Colab notebook](https://colab.research.google.com/drive/1shivu4GtUoiDzDrE9RKD1RuEm3OqXJuD?usp=sharing) that contains all the steps needed to reproduce the results directly within Colab.

**Comment by BhLee98** [comment]
Thanks, I still have some confusion about the loss of **Contrastive pretraining** training. may I ask in which file this part of the code is located? Does every point in the volume participate in the contrastive  learning calculation during the actual computation? I guess this may consume large amount of memory. 
Thanks anyway.

**Comment by neel-dey** [comment]
No, we sample 512 spatial indices at random for each decoder layer at each training iteration (Sec3/"Pretraining Implementation Details").

The contrastive pretraining portion of the code isn't out yet but will be before the conference. If you're interested in doing it yourself, it's simply a voxel- and layer-wise version of [supervised contrastive learning](https://github.com/HobbitLong/SupContrast) on randomly sampled spatial indices.

**Comment by neel-dey** [comment]
> Thanks, I still have some confusion about the loss of **Contrastive pretraining** . may I ask in which file this part of the code is located? Does every point in the volume participate in the contrastive learning calculation during the actual computation? I guess this may consume large amount of memory. Thanks anyway.

@BhLee98 if you're still interested in the contrastive pretraining code, it is now available here: https://github.com/neel-dey/anatomix/tree/main/pretraining (sorry for the long wait!)
